freedomofpress/securedrop
# Merge qa_loader and create-dev-data; add seen data.

## Status

Ready for review

## Description of Changes

Fixes #5514.

There was a lot of overlap between `qa_loader.py` and `create-dev-data.py`, so we're consolidating them while adding seen data to the populated database.

## Testing

- Run `make dev` and verify that the database is populated correctly.
- Run `make build-debs && make staging`, log in to the app server, and run:
  - `sudo -u www-data bash`
  - `cd /var/www/securedrop`
  - `CRYPTOGRAPHY_ALLOW_OPENSSL_102=yes ./loaddata.py --source-count ALL`. 
  The default three journalists and a number of sources should be created without any errors.
  - Run `loaddata.py` a few more times, varying the number of journalists, sources, and files, messages and replies per source, to be created. Confirm the results are as expected.

## Deployment

This is dev/test only.

## Checklist

### If you made changes to the server application code:


### If you made non-trivial code changes:


Choose one of the following:

- [x] I have opened https://github.com/freedomofpress/securedrop-docs/pull/108 in the [docs repo](https://github.com/freedomofpress/securedrop-docs) for these changes
